src/agent/agent.c

```diff
@@ -1038,6 +1045,81 @@ static const sd_bus_vtable internal_agent_vtable[] = {
         SD_BUS_VTABLE_END
 };
 
+/*************************************************************************
+ ********** org.containers.hirte.Agent.CreateProxy ***********************
+ ************************************************************************/
+static int agent_method_create_proxy(UNUSED sd_bus_message *m, void *userdata, UNUSED sd_bus_error *ret_error) {
+        Agent *agent = (Agent *) userdata;
+        if (agent == NULL) {
+                hirte_log_error("Failed to create proxy. Agent is NULL.");
+                return EINVAL;
+        }
+
+        _cleanup_sd_bus_message_ sd_bus_message *reply = NULL;
+
+        const char *node_name = NULL;
+        const char *unit_name = NULL;
+        const char *mode = NULL;
+        int r = sd_bus_message_read(m, "sss", &node_name, &unit_name, &mode);
+        if (r < 0) {
+                return sd_bus_reply_method_errorf(m, SD_BUS_ERROR_INVALID_ARGS, "Invalid arguments");
+        }
+
+        hirte_log_debugf(
+                        "Received create proxy request for node %s on %s with mode %s",
+                        node_name,
+                        unit_name,
+                        mode);
+
+        _cleanup_proxy_job_ ProxyJob *proxy = proxy_job_new(agent, node_name, unit_name, mode, m);
+        if (proxy == NULL) {
+                return sd_bus_reply_method_errorf(reply, SD_BUS_ERROR_FAILED, "Internal error");
+        }
+
+        if (!proxy_job_export(proxy)) {
+                return sd_bus_reply_method_errorf(reply, SD_BUS_ERROR_FAILED, "Internal error");
+        }
+
+        LIST_APPEND(proxy_jobs, agent->outstanding_proxy_jobs, proxy_job_ref(proxy));
+
+        r = proxy_job_emit_proxy_new(proxy);
+        if (r < 0) {
+                hirte_log_errorf("Failed to emit ProxyNew signal: %s", strerror(-r));
+                return sd_bus_reply_method_errorf(m, SD_BUS_ERROR_FAILED, "Internal error");
```

Comment:
This isn't right I think. We should either report an error to the, and then *not* add the job to the list, or alternatively, not report an error if we added the proxy to the list.

I think the later is maybe the correct approach. What if the agent is not yet connected at the moment, but later connects with the manager. We then re-send the ProxyNew, and will eventually get a reply.

Reply:
Actually maybe we should do both. Fail the call if any random error, but silently succeed if emit_proxy_new() returns the specific error for "not online yet".

Reply:
(Alternatively do the is-online check before calling emit)